### ISSUE TITLE
tcmalloc free error when building project

### ISSUE THREAD

**Issue by robinp** [commercialhaskell/stack#2926]
I minimalised an example that reproduces an exception during build. Please help me zoom on the root cause.

Stack version string is: `Version 1.3.2 x86_64 hpack-0.15.0`. `stack exec -- ghc --version` is `The Glorious Glasgow Haskell Compilation System, version 7.10.3`.

When running

    ~/.local/bin/stack build -v --dump-logs > out 2> err

Output:

```
...
2017-01-15 13:19:20.858672: [debug] Run process: /home/ron/.stack/setup-exe-cache/x86_64-linux/Cabal-simple_mPHDZzAJ_1.22.5.0_ghc-7.10.3 --builddir=.stack-work/dist/x86_64-linux/Cabal-1.22.5.0 build lib:x --ghc-options " -ddump-hi -ddump-to-file"
@(System/Process/Read.hs:306:3)
2017-01-15 13:19:20.898025: [info] Preprocessing library x-0.1.0.0...
@(Stack/Build/Execute.hs:1013:67)
2017-01-15 13:19:20.971815: [info] [1 of 1] Compiling Lib              ( src/Lib.hs, .stack-work/dist/x86_64-linux/Cabal-1.22.5.0/build/Lib.o )
@(Stack/Build/Execute.hs:1013:67)
2017-01-15 13:19:21.189262: [warn] src/tcmalloc.cc:283] Attempt to free invalid pointer 0x1d97540
@(Stack/Build/Execute.hs:1013:67)
2017-01-15 13:19:21.344372: [debug] Start: getPackageFiles /tmp/test/x/x.cabal
@(Stack/Package.hs:250:21)
2017-01-15 13:19:21.345232: [debug] Finished in 0ms: getPackageFiles /tmp/test/x/x.cabal
```

Env: Arch Linux 64 bit, project fully built with stack (lts-6.27 and lts-7.15 behaves the same). Arch's `extra/leveldb` is version `1.18-2`. Note that the project just depends on leveldb, but does not use it at all. Strangely the dependency is needed to trigger the bug. I tried on two different machines with Arch and it happened - no idea about other distros.

## Reproduction

See https://gist.github.com/robinp/184fdbe1a4d199601c444439e7c6464e for the files.

TLDR:

### x.cabal:

```
build-depends: base
, leveldb-haskell
, neat-interpolation
, text-format
```
Note that neither leveldb-haskell, nor text-format are actually used, but removing any stops the bug from happening.

### stack.yaml:
```
# resolver: lts-7.15
resolver: lts-6.27

packages:
- 'x'
```
###  src/Lib.hs

```
{-# LANGUAGE QuasiQuotes #-}
module Lib where

import NeatInterpolation

renderXmlRequest = [text|apple|]
```

Then do `stack build`. Interestingly, doing `stack exec -- ghc x/src/Lib.hs` builds the Lib fine, so I'm not sure where the error with stack build is coming from.

Thank you!

**Comment by robinp** [comment]
Couldn't reproduce in Docker. Closing for now.

**Comment by xavier83** [comment]
I am still experiencing this on Arch but when compiling https://github.com/lamdu/lamdu/issues/51
